python/lsst/daf/butler/core/webdavutils.py

```diff
@@ -0,0 +1,167 @@
+    except KeyError:
+        raise KeyError("Environment variable WEBDAV_AUTH_METHOD is not set, please use values X509 or TOKEN")
+
+    if env_auth_method == "X509":
+        try:
+            proxy_cert = os.environ['WEBDAV_PROXY_CERT']
+        except KeyError:
+            raise KeyError("Environment variable WEBDAV_PROXY_CERT is not set")
+        s.cert = (proxy_cert, proxy_cert)
+    elif env_auth_method == "TOKEN":
+        try:
+            bearer_token = os.environ['WEBDAV_BEARER_TOKEN']
+        except KeyError:
+            raise KeyError("Environment variable WEBDAV_BEARER_TOKEN is not set")
+        s.headers = CaseInsensitiveDict({'Authorization': 'Bearer ' + bearer_token})
+    else:
+        raise ValueError("Environment variable WEBDAV_AUTH_METHOD must be set to X509 or TOKEN")
+
+    # s.verify = False
+    log.debug("Session configured and ready.")
+
+    return s
+
+
+def webdavCheckFileExists(path: Union[Location, ButlerURI, str],
+                          session: Optional[requests.Session] = None) -> Tuple[bool, int]:
+    """Check that a remote HTTP resource exists."""
+    if session is None:
+        session = getHttpSession()
+
+    filepath = getFileURL(path)
+
+    log.debug("Checking if file exists: %s", filepath)
+
+    r = session.head(filepath)
+    return (True, int(r.headers['Content-Length'])) if r.status_code == 200 else (False, -1)
+
+
+def webdavDeleteFile(path: Union[Location, ButlerURI, str],
+                     session: Optional[requests.Session] = None) -> None:
+    """Check that a remote HTTP resource exists."""
+    if session is None:
+        session = getHttpSession()
+
+    filepath = getFileURL(path)
+
+    log.debug("Removing file: %s", filepath)
+    r = session.delete(filepath)
+    if r.status_code not in [200, 202, 204]:
+        raise FileNotFoundError(f"Unable to delete resource {filepath}; status code: {r.status_code}")
+
+
+def folderExists(path: Union[Location, ButlerURI, str],
+                 session: Optional[requests.Session] = None) -> bool:
+    """Check if the Webdav repository at a given URL actually exists.
+
+    Parameters
+    ----------
+    folderName : `str`
+        Name of the Webdav folder
+    session : `requests.Session`, optional
+        Session object to query.
+
+    Returns
+    -------
+    exists : `bool`
+        True if it exists, False if no folder is found.
+    """
+    if session is None:
+        session = getHttpSession()
+
+    filepath = getFileURL(path)
+
+    log.debug("Checking if folder exists: %s", filepath)
+    r = session.head(filepath)
+    return True if r.status_code == 200 else False
+
+
+def isWebdavEndpoint(path: Union[Location, ButlerURI, str]) -> bool:
+
+    filepath = getFileURL(path)
+
+    log.debug("Detecting HTTP endpoint type...")
+    r = requests.options(filepath)
+    return True if 'DAV' in r.headers else False
+
+
+def getFileURL(path: Union[Location, ButlerURI, str]) -> str:
```

Comment:
I think I'd keep this as a private function and not export it. I'd consider dropping `Location` support and just make callers use `location.uri`.

The str check isn't needed because in theory you can always do `ButlerURI(path).geturl()` for str and ButlerURI.

Reply:
The problem with dropping `Location` support in `getFileURL` is that the callers aren't aware of the type of the `path` they send when calling `getFileURL`.
This means that we should either completely drop `Location` support in callers also, and make sure themselves are never called with a `Location` parameter somewhere higher in the code, **or** do the `isInstance(path, Location)` test every time before we call `getFileURL`, which is probably not desirable.

Reply:
Ok, this will go away when I do a clean up pass I think. Please make it a private API though.